sql/src/main/java/io/crate/execution/engine/collect/stats/JobsLogs.java

```diff
@@ -144,13 +151,14 @@ private void addToHistogram(JobContextLog log) {
      */
     public void logPreExecutionFailure(UUID jobId, String stmt, String errorMessage, User user) {
         JobContextLog jobContextLog = new JobContextLog(
-            new JobContext(jobId, stmt, System.currentTimeMillis(), user, null), errorMessage);
+            new JobContext(jobId, stmt, System.currentTimeMillis(), user, new StatementClassifier.Classification(UNDEFINED)), errorMessage);
         jobsLogRWLock.readLock().lock();
         try {
             jobsLog.add(jobContextLog);
         } finally {
             jobsLogRWLock.readLock().unlock();
         }
+        addToHistogram(jobContextLog);
```

Comment:
I'd like to rename this from `addToHistogram` to `recordStatementMetrics` (or something similar)

Reply:
How about just `recordMetrics` ? (We in a sense have no "not statement" metrics)